Products/CMFPlone/browser/recyclebin.py

```diff
@@ -0,0 +1,622 @@
+from datetime import datetime
+from Products.CMFCore.utils import getToolByName
+from Products.CMFPlone.interfaces.recyclebin import IRecycleBin
+from Products.Five.browser import BrowserView
+from Products.Five.browser.pagetemplatefile import ViewPageTemplateFile
+from Products.statusmessages.interfaces import IStatusMessage
+from z3c.form import button
+from z3c.form import field
+from z3c.form import form
+from zExceptions import NotFound
+from zope import schema
+from zope.component import getUtility
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.publisher.interfaces import IPublishTraverse
+
+import logging
+import uuid
+
+
+logger = logging.getLogger(__name__)
+
+
+# Utility functions to avoid code duplication
+def format_date(context, date):
+    """Format date for display"""
+    if date is None:
+        return ""
+    portal = getToolByName(context, "portal_url").getPortalObject()
+    return portal.restrictedTraverse("@@plone").toLocalizedTime(date, long_format=True)
+
+
+def format_size(size_bytes):
+    """Format size in bytes to human-readable format"""
+    if size_bytes < 1024:
+        return f"{size_bytes} B"
+    elif size_bytes < 1024 * 1024:
+        return f"{size_bytes / 1024:.1f} KB"
+    else:
+        return f"{size_bytes / (1024 * 1024):.1f} MB"
+
+
+class IRecycleBinForm(Interface):
+    """Schema for the recycle bin form"""
+
+    selected_items = schema.List(
+        title="Selected Items",
+        description="Selected items for operations",
+        value_type=schema.TextLine(),
+        required=False,
+    )
```

Comment:
In general we have moved all interfaces and settings schematas to `plone.base.interfaces.controlpanel` ... this should be refactored the same. And rename `IReciclyBinForm` to `IRecycleBinSettings`.

Reply:
done in https://github.com/plone/Products.CMFPlone/pull/4168/commits/2ed81c8fcbf8a02f5533409b82c34aa0a6818835